From: Liu Bo <liubo2009@cn.fujitsu.com>
Date: Fri, 29 Jun 2012 03:58:47 -0600
Patch-mainline: 3.6
Git-commit: 768e9dfe820abdcfb6683e05c60b8634f1a4ffce
References: FATE#312888
Subject: [PATCH] Btrfs: remove redundant r/o check for superblock

mnt_want_write() and mnt_want_write_file() will check sb->s_flags with
MS_RDONLY, and we don't need to do it ourselves.

Signed-off-by: Liu Bo <liubo2009@cn.fujitsu.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c |    8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1388,15 +1388,11 @@ static noinline int btrfs_ioctl_snap_cre
 						    u64 *transid,
 						    bool readonly)
 {
-	struct btrfs_root *root = BTRFS_I(fdentry(file)->d_inode)->root;
 	struct file *src_file;
 	int namelen;
 	int ret = 0;
 
-	if (root->fs_info->sb->s_flags & MS_RDONLY)
-		return -EROFS;
-
-	ret = mnt_want_write_file(file);
+	ret = mnt_want_write(file->f_path.mnt);
 	if (ret)
 		goto out;
 
@@ -3279,7 +3275,7 @@ static long btrfs_ioctl_balance(struct f
 	if (fs_info->sb->s_flags & MS_RDONLY)
 		return -EROFS;
 
-	ret = mnt_want_write_file(file);
+	ret = mnt_want_write(file->f_path.mnt);
 	if (ret)
 		return ret;
 
